src/hyphenClient.ts

```diff
@@ -0,0 +1,74 @@
+import type { EvaluationResponse, HyphenEvaluationContext } from './types';
+import NodeCache from '@cacheable/node-cache';
+import { horizon, cache } from './config';
+
+export class HyphenClient {
+  private readonly publicKey: string;
+  private readonly horizonServerUrls: string[];
+  private cache: NodeCache;
+
+  constructor(publicKey: string, horizonServerUrls?: string[]) {
+    this.publicKey = publicKey;
+    this.horizonServerUrls = this.getServerUrls(horizonServerUrls);
+    this.cache = new NodeCache({
+      stdTTL: cache.ttl,
+      checkperiod: cache.ttl * 2,
+    });
+  }
+
+  async evaluate(context: HyphenEvaluationContext): Promise<EvaluationResponse> {
+    const cachedResponse = this.cache.get<EvaluationResponse>(context.targetingKey);
+    if (cachedResponse) {
+      return cachedResponse;
+    }
+
+    const evaluationResponse = await this.fetchEvaluationResponse(this.horizonServerUrls, context);
+
+    if (evaluationResponse) {
+      this.cache.set(context.targetingKey, evaluationResponse);
+    }
+    return evaluationResponse;
+  }
+
+  private getServerUrls(horizonServerUrls: string[] = []) {
+    if (!horizonServerUrls.includes(horizon.url)) {
+      horizonServerUrls.push(horizon.url);
+    }
+    return horizonServerUrls;
+  }
```

Comment:
should be a static function. 

you don't really need to check for the url you can just tack it on. it's a small thing but this happens in the ctor and you want that to be fast.